script/deployPoolWithRenzoStabilityAndMintLiquidity.s.sol

```diff
@@ -0,0 +1,241 @@
+            POOLMANAGER,
+            rateProvider,
+            minFee,
+            maxFee,
+            ezETH
+        );
+        (address hookAddress, bytes32 salt) = HookMiner.find(
+            CREATE2_FACTORY,
+            flags,
+            type(RenzoStability).creationCode,
+            constructorArgs
+        );
+
+        startingPrice = SqrtPriceLibrary.exchangeRateToSqrtPriceX96(
+            rateProvider.getRate()
+        );
+        // Deploy the hook using CREATE2
+        vm.startBroadcast();
+        RenzoStability renzoStability = new RenzoStability{salt: salt}(
+            POOLMANAGER,
+            rateProvider,
+            minFee,
+            maxFee,
+            ezETH
+        );
+        vm.stopBroadcast();
+        // check that the hook was deployed at the expected address
+        require(
+            address(renzoStability) == hookAddress,
+            "RenzoStability: hook address mismatch"
+        );
+
+        // log the address of the deployed hook
+        console2.log("RenzoStability hook deployed at - ");
+        console2.logAddress(address(renzoStability));
+        return address(renzoStability);
+    }
+
+    /// @dev helper function for encoding mint liquidity operation
+    /// @dev does NOT encode SWEEP, developers should take care when minting liquidity on an ETH pair
+    function _mintLiquidityParams(
+        PoolKey memory poolKey,
+        int24 _tickLower,
+        int24 _tickUpper,
+        uint256 liquidity,
+        uint256 amount0Max,
+        uint256 amount1Max,
+        address _recipient,
+        bytes memory hookData
+    ) internal pure returns (bytes memory, bytes[] memory) {
+        bytes memory actions = abi.encodePacked(
+            uint8(Actions.MINT_POSITION),
+            uint8(Actions.SETTLE_PAIR),
+            uint8(Actions.SWEEP)
+        );
+
+        bytes[] memory params = new bytes[](3);
+        params[0] = abi.encode(
+            poolKey,
+            _tickLower,
+            _tickUpper,
+            liquidity,
+            amount0Max,
+            amount1Max,
+            _recipient,
+            hookData
+        );
+        params[1] = abi.encode(poolKey.currency0, poolKey.currency1);
+        params[2] = abi.encode(poolKey.currency0, _recipient);
+        return (actions, params);
+    }
+
+    function tokenApprovals() public {
+        if (!currency0.isAddressZero()) {
+            token0.approve(address(PERMIT2), type(uint256).max);
+            PERMIT2.approve(
+                address(token0),
+                address(posm),
+                type(uint160).max,
+                type(uint48).max
+            );
+        }
+        if (!currency1.isAddressZero()) {
+            token1.approve(address(PERMIT2), type(uint256).max);
+            PERMIT2.approve(
+                address(token1),
+                address(posm),
+                type(uint160).max,
+                type(uint48).max
+            );
+        }
+    }
```

Comment:
this logic is safe and works as intended, but max approvals in a production script feel a bit scary

Reply:
added tokenAmounts instead of max value approvals